core/src/main/java/org/apache/calcite/plan/volcano/TopDownRuleDriver.java

```diff
@@ -69,8 +69,8 @@
   private GeneratorTask applying = null;
 
   /**
-   * RelNodes that are generated by passThrough or derive
-   * these nodes will not takes part in another passThrough or derive.
+   * RelNodes that are generated by passThrough or derive.
+   * These nodes will not take part in another passThrough or derive.
```

Comment:
It would be helpful to add some links to the `passThrough` and `derive` operations.

Reply:
Sounds reasonable.